libs/MVS/SceneTexture.cpp

```diff
@@ -509,11 +517,34 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 		faceMap.create(imageData.height, imageData.width);
 		depthMap.create(imageData.height, imageData.width);
 		RasterMesh rasterer(vertices, imageData.camera, depthMap, faceMap);
+		// creating mask for the image border
+		rasterer.mask = Image8U(imageData.height + 2, imageData.width + 2);
+		rasterer.mask = rasterer.mask != rasterer.mask;
+		Image8U imageCopy;
+		cv::cvtColor(imageData.image, imageCopy, cv::COLOR_BGR2GRAY);
+		imageCopy -= 1; // this line gets rid of white spots in the middle of the image
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, 0), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(0, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows - 1), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, imageCopy.rows / 2), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols - 1, 0), 255);
+		cv::floodFill(imageCopy, rasterer.mask, cv::Point(imageCopy.cols / 2, 0), 255);
+		cv::Mat kernel = cv::getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3));
+		cv::erode(rasterer.mask, rasterer.mask, kernel);
+		rasterer.mask = rasterer.mask == 0;
```

Comment:
why do you invert the values here, when you could simply initialize it with 255 and set 0 in cv::floodfill?

Reply:
as a parameter convention the mask has to be initialized with 0 so that floodfill will work

Reply:
since the mask is a binary matrix it would fill with one the pixels that floodfill changes in the original image, and leave the unchanged ones with 0

Reply:
interesting, I didn't know about all these conventions, thx